### ISSUE TITLE
EMI Integration is hardcoded

### ISSUE THREAD

**Issue by meeprdib** [AstralOrdana/Spelunkery#295]
**Describe the bug:**
A clear and concise description of what the bug is. Include as much information as you can. Images are often helpful, especially screenshots of the F3 debug menu.

I have tried to remove Sulfur from being obtained from Sluices via datapacks, but the recipes remain unaffected. The datapack filepath is `kubejs\data\spelunkery\loot_tables\gameplay\sluice\spring_water\passive.json`, and the json reads as follows:
```
{
  "pools": [
    {
      "rolls": 1,
      "bonus_rolls": 0,
      "entries": [
        {
          "type": "minecraft:item",
          "weight": 15,
          "name": "spelunkery:salt"
        },
        {
          "type": "minecraft:item",
          "weight": 15,
          "name": "spelunkery:saltpeter"
        },
        {
          "type": "minecraft:item",
          "weight": 5,
          "name": "spelunkery:raw_copper_nugget"
        },
        {
          "type": "minecraft:item",
          "weight": 5,
          "name": "spelunkery:raw_iron_nugget"
        }
      ]
    }
  ]
}
```

However, despite Sulfur not being included in the loot table, it stills shows up as an output in JEI. 
![image](https://github.com/user-attachments/assets/2869ed93-6205-478c-a779-6b1ecb65dc8b)

**Versions: (BEFORE SUBMITTING A BUG REPORT, make sure you have the most up-to-date versions of Spelunkery, Moonlight Lib and Supplementaries)**
Minecraft version: 1.20.1
Spelunkery version: 1.20.1-0.3.13
Moonlight Lib version: 1.20-2.13.21
Fabric API / QSL / Forge version: 47.3.0

**Other mods:**
List all mods that you're using; could be in the form of a list of the filenames, a crash report list, a screenshot of
the mods folder, etc. Just make sure all mods used are known!

[OdysseyNexus Dev.txt](https://github.com/user-attachments/files/17626985/OdysseyNexus.Dev.txt)

**Logs:**
*Even if the game does not crash they may be useful!*
Please provide the following logs:
From .minecraft/logs :  "latest.log" as well as "debug.log" if it exists. From .minecraft : "launcher_log" as well as a
file that starts with "hs_err" if it exists. Note: Double check that the logs are from the correct time. ex, don't give
us a log from a month ago because you forgot to check the date.

[latest.log](https://github.com/user-attachments/files/17626998/latest.log)

**To Reproduce:**
Steps to reproduce the behavior. Format it as a numbered list of steps.

1. Make a datapack overriding base Spelunkerys Sluice recipe
2. Reload game
3. Check JEI

**Expected behavior:**

I expected Sulfur to not show up in any World Interaction / Sluice recipes after I removed them with a datapack.

Edit: Fixed datapack filepath

**Comment by tehgreatdoge** [comment]
It should be in kubejs/data/spelunkery instead of kubejs/spelunkery.

**Comment by tehgreatdoge** [comment]
Also, I think the loot tables for passive sluicing for fluids with underscores in their names are broken? (See: #285 and the issue that links to it)
There is a temporary workaround where you can remove the underscores in the name of the fluid in the path, so that it becomes "/springwater/" instead of "/spring_water/"

**Comment by meeprdib** [comment]
> It should be in kubejs/data/spelunkery instead of kubejs/spelunkery.

Ah, I had that as the filepath, just forgot to put it in this issue. Thanks for the catch.

> Also, I think the loot tables for passive sluicing for fluids with underscores in their names are broken? (See: #285 and the issue that links to it) There is a temporary workaround where you can remove the underscores in the name of the fluid in the path, so that it becomes "/springwater/" instead of "/spring_water/"

I will try this out.

**Comment by meeprdib** [comment]
That doesn't work for me. Filepath is `kubejs\data\spelunkery\loot_tables\gameplay\sluice\springwater\passive.json` and recipes are still unaffected.

**Comment by tehgreatdoge** [comment]
Yes, but have you tried using the actual sluice? JEI shows the correct loot table (data\spelunkery\loot_tables\gameplay\sluice\spring_water\passive.json), however, due to the bug, the path that the sluice actually uses for passive recipes is data\spelunkery\loot_tables\gameplay\sluice\springwater\passive.json (which, results in passive sluicing being broken for fluids with an underscore in their name (spring_water and portal_fluid))

**Comment by tehgreatdoge** [comment]
Oh nevermind lol, apparently the EMI integration is hardcoded lol (either way, try using the sluice itself)

**Comment by meeprdib** [comment]
:unamused:

**Comment by meeprdib** [comment]
Changing the issue title as this is the core problem behind this. Assuming the JEI integration isn't hardcoded as well (haven't tried it out at all), I'd suggest just removing the EMI "compat" in its entirety, considering that EMI is meant to be used alongside JEI. Unless I'm missing something here, of course.

**Comment by dandalous** [comment]
Not really. EMI is meant to be standalone. JEI is just so ubiquitous that EMI has built in compatibility for most other mods using JEI. Slowly but surely I've been seeing native EMI support with major mods like Create and Refined Storage

**Comment by alperen-bircak** [comment]
I just stumbled upon this issue myself, would be great for modpack developers if it is fixed.
